src/appleseed.studio/mainwindow/rendering/renderingmanager.cpp

```diff
@@ -490,6 +495,105 @@ void RenderingManager::slot_camera_change_end()
 {
 }
 
+void RenderingManager::slot_material_dropped(const foundation::Vector2d& drop_pos, const std::string& material_name)
+{
+    if (!m_project->has_trace_context())
+    {
+        RENDERER_LOG_INFO("the scene must be rendering or must have been rendered at least once for drag&drop to be available.");
+        return;
+    }
+
+    DroppedMaterialMenu* menu = new DroppedMaterialMenu(drop_pos, material_name);
```

Comment:
Do you confirm that the menu is destroyed automatically?

Reply:
I just made a test project and it is not
we could define menu as a static variable to prevent appleseed from building one each time you drop an item.

Reply:
Maybe it's not destroyed automatically because you're not passing a parent.

Reply:
I did some test with a dummy project but it keep crashing but it seems to work fine on appleseed
I added this 2 lines right after each signal_apply_material in droppedmaterialmenu.cpp
`destroy();`
`delete this;`
Do you have idea to test the stability of this solution?

Reply:
I don't think you want to do any of this. Just passing a parent to `DroppedMaterialMenu` and eventually to `QMenu` should fix any ownership issue.

Reply:
I tried with passing the mainwindow as a parent, but it doesnt clean the menu
If you pass a parent, it will only clean the widget when the parent will be destroy:

> Although a popup menu is always a top-level widget, if a parent is passed the popup menu will be deleted when that parent is destroyed

Reply:
Some hints about possible solutions here:
https://stackoverflow.com/questions/39259862/when-and-how-to-properly-destroy-qmenu-context-menu

Make sure to check out both solutions and read all comments of the first solution.

I kind of like the second solution.